hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AInputStream.java

```diff
@@ -561,6 +566,12 @@ public synchronized void close() throws IOException {
       try {
         // close or abort the stream; blocking
         awaitFuture(closeStream("close() operation", false, true));
+        // Collect ThreadLevel IOStats
+        if (ioStatisticsContext != null) {
+          ioStatisticsContext.getThreadIOStatistics().aggregate(streamStatistics.getIOStatistics());
+          LOG.debug("IOStatistics of thread: {}\n{}",
+              Thread.currentThread().getId(), ioStatisticsContext.toString());
```

Comment:
too expensive. 
Use IOStatisticsLogging for on-demand logging of the thread io stats you'll be caching